samples/nrf9160/cloud_simple/src/main.c

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <misc/reboot.h>
+#include <logging/log_ctrl.h>
+#include <lte_lc.h>
+#include <net/cloud.h>
+#include <net/socket.h>
+#include <dk_buttons_and_leds.h>
+
+static struct cloud_backend *cloud_backend;
+static struct k_delayed_work cloud_update_work;
+
+enum error_type {
+	ERROR_CLOUD,
+	ERROR_BSD_RECOVERABLE,
+	ERROR_LTE_LC,
+	ERROR_SYSTEM_FAULT
+};
+
+/**@brief Cloud simple error handler. */
+void error_handler(enum error_type err_type, int err_code)
+{
+	switch (err_type) {
+	case ERROR_CLOUD:
+		printk("Error of type ERROR_CLOUD: %d\n", err_code);
+		break;
+	case ERROR_BSD_RECOVERABLE:
+		printk("Error of type ERROR_BSD_RECOVERABLE: %d\n", err_code);
+		break;
+	case ERROR_LTE_LC:
+		printk("Error of type ERROR_LTE_LC: %d\n", err_code);
+		break;
+	case ERROR_SYSTEM_FAULT:
```

Comment:
As a sample, you shouldn't need a custom system fault handler, and I am not convinced that you should reboot on fault instead of looping either.